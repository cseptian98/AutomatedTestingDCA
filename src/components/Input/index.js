import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';
import axiosConfig from '../../api/BaseConfig';

const InputTodo = () => {
  const [title, setTitle] = useState('');

  const doCreate = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
    };

    const onFailure = error => {
      console.log('debug error', error.response.data);
    };

    axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure);
  }

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={doCreate}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 20,
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
