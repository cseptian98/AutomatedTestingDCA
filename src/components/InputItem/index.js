import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import axiosConfig from 'api/BaseConfig';
import styles from './InputItems.styles';

const InputItem = () => {
  const listId = 1;
  const [title, setTitle] = useState('');

  const createNewItem = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
      setTitle('');
    };

    const onFailure = error => {
      console.log('debug error', error);
    };

    axiosConfig
      .post('api/TodoItems', {listId, title})
      .then(onSuccess)
      .catch(onFailure);
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo Item..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={createNewItem}>
        <Icon name="add" size={24} />
      </TouchableOpacity>
    </View>
  );
};

export default InputItem;
