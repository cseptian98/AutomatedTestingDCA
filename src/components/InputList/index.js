import React, {useState} from 'react'
import {View, TextInput, TouchableOpacity, Image} from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './InputList.styles'
import axiosConfig from 'api/BaseConfig'

const InputList = ({createList}) => {
  const [title, setTitle] = useState('')

  // const createNewList = () => {
  //   const onSuccess = ({data}) => {
  //     console.log('debug success', data)
  //     setTitle('')
  //   }

  //   const onFailure = error => {
  //     console.log('debug error', error)
  //   }

  //   axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure)
  // }

  return (
    <View style={styles.container}>
      <View>
        <Image
          style={styles.image}
          source={require('assets/images/list.png')}
        />
      </View>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#FFF"
          onChangeText={title => setTitle(title)}
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => createList(title)}>
          <Icon name="add" size={30} color="#FFF" />
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default InputList
