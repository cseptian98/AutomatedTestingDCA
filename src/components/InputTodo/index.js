import React from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';

const InputTodo = () => {
  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
        />
      </View>
      <TouchableOpacity style={styles.button}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
