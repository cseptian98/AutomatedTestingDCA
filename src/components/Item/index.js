import React from 'react'
import {TouchableOpacity, Text} from 'react-native'
import styles from './Item.styles'

const Item = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
)

export default Item
