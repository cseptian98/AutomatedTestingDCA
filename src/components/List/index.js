import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';
import styles from './List.styles';

const List = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

export default List;
