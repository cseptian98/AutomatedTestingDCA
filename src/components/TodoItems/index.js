import React, {useState, useEffect} from 'react';
import {SafeAreaView, FlatList, StyleSheet} from 'react-native';
import List from 'components/List';
import axiosConfig from 'api/BaseConfig';

const TodoItems = () => {
  const ListId = 1;
  const [item, setItem] = useState([]);

  useEffect(() => {
    getTodoItems();
  }, []);

  const getTodoItems = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data.items);
      setItem(data.items);
    };

    const onFailure = error => {
      console.log('debug error', error.response.data);
    };

    axiosConfig
      .get('api/TodoItems', {params: {ListId}})
      .then(onSuccess)
      .catch(onFailure);
  };

  const renderItem = ({item}) => <List title={item.title} />;

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={item}
        renderItem={renderItem}
        keyExtractor={item => item.id}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFF',
  },
});

export default TodoItems;
