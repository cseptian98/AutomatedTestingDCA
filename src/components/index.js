import FormUpdateItem from './TodoItems/FormUpdateItem/FormUpdateItem'
import TodoLists from './TodoLists/TodoLists'
import TodoItems from './TodoItems/TodoItems'
import User from './User/User'

export {FormUpdateItem, TodoLists, TodoItems, User}