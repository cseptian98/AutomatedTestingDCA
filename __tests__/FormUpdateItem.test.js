import React from 'react'
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import {FormUpdateItem} from 'components'
import {
  TEST_ID_IMAGE_UPDATE_ITEM,
  TEST_ID_NOTES_ITEM,
} from 'constants'

let componentFromUpdate
let inputNote
let notes
const params = {url: 'api/TodoItems/UpdateItemDetails'}

describe('From Update Item', () => {
  beforeEach(() => {
    componentFromUpdate = render(<FormUpdateItem route={{params}} />)
    notes = 'Note'
    inputNote = componentFromUpdate.getByTestId(TEST_ID_NOTES_ITEM)
  })
  it('renders correctly', async () => {
    await waitFor(() => {
      const screen = componentFromUpdate.toJSON()
      expect(screen).toMatchSnapshot()
    })
  })
  it('should show form update item screen', () => {
    expect(
      componentFromUpdate.getByTestId(TEST_ID_IMAGE_UPDATE_ITEM),
    ).toBeTruthy()
  })
  it('should change text input value', async () => {
    await waitFor(() => {
      fireEvent.changeText(inputNote, notes)
    })
    expect(inputNote.props.value).toBe(notes)
  })
})
