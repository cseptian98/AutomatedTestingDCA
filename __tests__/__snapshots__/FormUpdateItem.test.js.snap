// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From Update Item renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/update_item.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginBottom": 16,
        "marginTop": 24,
        "width": 280,
      }
    }
    testID="TEST_ID_IMAGE_UPDATE_ITEM"
  />
  <View
    style={
      Object {
        "borderBottomColor": "#212121",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Note"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="TEST_ID_NOTES_ITEM"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0D47A1",
        "borderRadius": 16,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="TEST_ID_BUTTON_UPDATE_DETAIL"
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 16,
        }
      }
    >
      Update Item
    </Text>
  </View>
</View>
`;
